File: backend/app/schemas/orgchart.py
from pydantic import BaseModel, ConfigDict


class OrgchartPerson(BaseModel):
    employee_id: int
    external_id: str
    first_name: str
    last_name: str

    profession_type: str | None = None
    org_role: str | None = None
    is_executive: bool = False

    email: str | None = None
    phone: str | None = None
    employment_type: str | None = None
    weekly_hours: float | None = None

    model_config = ConfigDict(from_attributes=True)


class OrgchartLocationGroup(BaseModel):
    location_id: int
    location_name: str
    site_leads: list[OrgchartPerson]
    doctors: list[OrgchartPerson]
    mfas: list[OrgchartPerson]
    administration: list[OrgchartPerson]


class OrgchartResponse(BaseModel):
    executives: list[OrgchartPerson]
    locations: list[OrgchartLocationGroup]

File: frontend/src/api/orgchart.ts
import { api } from "./client";

export type OrgchartPerson = {
  employee_id: number;
  external_id: string;
  first_name: string;
  last_name: string;
  profession_type: string | null;
  org_role: string | null;
  is_executive: boolean;
  email: string | null;
  phone: string | null;
  employment_type: string | null;
  weekly_hours: number | null;
};

export type OrgchartLocationGroup = {
  location_id: number;
  location_name: string;
  site_leads: OrgchartPerson[];
  doctors: OrgchartPerson[];
  mfas: OrgchartPerson[];
  administration: OrgchartPerson[];
};

export type OrgchartResponse = {
  executives: OrgchartPerson[];
  locations: OrgchartLocationGroup[];
};

export async function fetchOrgchart(): Promise<OrgchartResponse> {
  return api<OrgchartResponse>("/api/orgchart");
}